src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -581,7 +581,7 @@ JvmtiVTMSTransitionDisabler::VTMS_vthread_unmount(jobject vthread, bool hide) {
     if (JvmtiExport::should_post_vthread_unmount()) {
       JvmtiExport::post_vthread_unmount(vthread);
     }
-    VTMS_unmount_begin(vthread);
+    VTMS_unmount_begin(vthread, /* last_unmount */ false);
```

Comment:
I think it would be better just to add 
thread->rebind_to_jvmti_thread_state_of(thread->threadObj());
after this call instead of adding a parameter to  VTMS_unmount_begin.
Just suggestion.

Reply:
So we need to execute the thread->rebind_to_jvmti_thread_state_of(thread->threadObj()) call after we do the cleanup to avoid removing the wrong state. So the other option to avoid the extra parameter would be to remove the rebind call from VTMS_unmount_begin(), which I also thought. But that would require to also add it to VTMS_vthread_unmount() for the normal unmount case, which already breaks the symmetry of all those methods, given that for the mount case the rebind call is encapsulated in VTMS_mount_end().

Reply:
Thanks for the explanation.